File: cpp/334.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        int val[2] = {INT_MAX, INT_MAX};
        for(auto& num : nums) {
            if(num<=val[0])
                val[0] = num;
            else if(num<=val[1])
                val[1] = num;
            else
                return true;
        }
        return false;
    }
};

File: python/334.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        val = [float('inf')]*2
        for num in nums:
            if(num<=val[0]):
                val[0] = num
            elif(num<=val[1]):
                val[1] = num
            else:
                return True
        return False